sdk/core/azure-core/test/ut/bodystream.cpp

```diff
@@ -33,7 +33,15 @@ class TestBodyStream : public BodyStream {
 TEST(BodyStream, Rewind)
 {
   TestBodyStream tb;
-  EXPECT_THROW(tb.Rewind(), std::logic_error);
+
+#if defined(NDEBUG)
+  // Release build won't provide assert msg
+  ASSERT_DEATH(tb.Rewind(), "");
+#else
+  ASSERT_DEATH(
+      tb.Rewind(),
+      "The specified BodyStream doesn't support Rewind which is required to guarantee fault ");
```

Comment:
This is begging for a new macro `ASSERT_DEATH_MEOW(x, y)` that expands to `ASSERT_DEATH(x, "")` when `defined(NDEBUG)` and to `ASSERT_DEATH(x, y)` otherwise.

Reply:
@vhvb1989 what do you think?